devito/types/dense.py

```diff
@@ -1399,7 +1402,7 @@ def __indices_setup__(cls, *args, **kwargs):
 
             if time_dim is None:
                 time_dim = grid.time_dim if isinstance(save, int) else grid.stepping_dim
-            elif not (isinstance(time_dim, Dimension) and time_dim.is_Time):
+            elif isinstance(time_dim, TimeDimension):
```

Comment:
`not`

Reply:
not Dimension and time_dim.is_Time

equivalent to 

TImeDimension

right?

Reply:
not Dimension and is_Time is impossible, check the parentheses it's not ( Dimension and tine)

Reply:
Sorry, yes. Interesting that no test failed! Reverted, as it is not semantically equivalent